src/app.py

```diff
@@ -983,26 +983,32 @@ def create_entity(entity_type):
 
         # Also check existence of the previous revision dataset if specified
         if 'previous_revision_uuid' in json_data_dict:
-            previous_version_dict = query_target_entity(json_data_dict['previous_revision_uuid'], user_token)
+            if isinstance(json_data_dict['previous_revision_uuid'], list):
+                previous_revision_list = json_data_dict['previous_revision_uuid']
 
-            # Make sure the previous version entity is either a Dataset or Sample (and publication 2/17/23)
-            if previous_version_dict['entity_type'] not in ['Sample'] and \
-                    not schema_manager.entity_type_instanceof(previous_version_dict['entity_type'], 'Dataset'):
-                bad_request_error(f"The previous_revision_uuid specified for this dataset must be either a Dataset or Sample or Publication")
+                nested_revisions =  app_neo4j_queries.nested_previous_revisions(neo4j_driver_instance, previous_revision_list)
+                if nested_revisions:
+                    bad_request_error(f"{nested_revisions[0][0]} is a revision of {nested_revisions[1][0]}. Datasets in previous_revision_uuid must not be revisions of eachother")
+            else:
+                previous_revision_list = [json_data_dict['previous_revision_uuid']]
+            for previous_revision in previous_revision_list:
+                previous_version_dict = query_target_entity(previous_revision, user_token)
+
+                # Make sure the previous version entity is either a Dataset or Sample (and publication 2/17/23)
+                if previous_version_dict['entity_type'] not in ['Sample'] and \
```

Comment:
Not introduced by your changes. But let's disallow `Sample` type, only allow `Dataset` (`Publication` too since it's a subclass of `Dataset`). We've never had Sample as the previous revision. And with the multi-revisions feature, it only applies to Dataset.